dev/benchmarks/macrobenchmarks/lib/src/web/bench_wrapbox_scroll.dart

```diff
@@ -97,9 +97,9 @@ class ProductPreview extends StatelessWidget {
           Container(
             margin: const EdgeInsets.all(23),
             padding: const EdgeInsets.all(18),
-            decoration: const BoxDecoration(
+            decoration: const ShapeDecoration(
```

Comment:
Dart fix should be able to make this change. There is a fairly new `replacedBy` rule that can replace one class type with another.
So I think in this case, the rule would want to make the change only when `BoxDecoration.shape` is set, have it `replacedBy` `ShapeDecoration`, and then change `shape` from `Box.circle` to `CircleBorder`.

You'll want to account for `BoxShape.rectangular` as well, since some folks may have it explicitly set in their code.

We have a lot of examples of these fix rules in  https://github.com/flutter/flutter/blob/master/packages/flutter/lib/fix_data.yaml, which is where these rules would be added as well.

There is also a wiki page that talks about dart fixes in greater detail: https://github.com/flutter/flutter/wiki/Data-driven-Fixes